presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -655,7 +650,30 @@ else if (type instanceof MapType) {
         Optional<Symbol> ordinalitySymbol = node.isWithOrdinality() ? Optional.of(unnestedSymbolsIterator.next()) : Optional.empty();
         checkState(!unnestedSymbolsIterator.hasNext(), "Not all output symbols were matched with input symbols");
 
-        UnnestNode unnestNode = new UnnestNode(idAllocator.getNextId(), projectNode, leftPlan.getFieldMappings(), unnestSymbols.build(), ordinalitySymbol, outer);
+        Optional<Expression> filterExpression = Optional.empty();
+        if (joinNode.getCriteria().isPresent()) {
+            JoinCriteria criteria = joinNode.getCriteria().get();
+            if (criteria instanceof JoinUsing || criteria instanceof NaturalJoin) {
+                throw notSupportedException(joinNode, "JOIN UNNEST with criteria other than ON");
```

Comment:
There's no such a thing as "JOIN UNNEST". It's a JOIN involving UNNEST (a collection-derived table, as the spec calls it). I'd change the message to "JOIN involving UNNEST with criteria ...." or even better, "Natural join involving UNNEST not supported", and "USING not supported for join involving UNNEST"